core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/internal/SdkClientConfigurationUtil.java

```diff
@@ -32,7 +32,8 @@
 import software.amazon.awssdk.metrics.MetricPublisher;
 import software.amazon.awssdk.profiles.ProfileFileSupplier;
 
-@SdkInternalApi
+// TODO(sra-plugins) Move this to codegen such that we can actually convert it to @SdkInternalApi
+@SdkProtectedApi
```

Comment:
Is this an upcoming PR/tracked somewhere?

Reply:
I added this one SMITHY-2106 to track it, but FWIW, I'm actively working on this change on top of this PR.